controllers/capabilities/tenant_internal_reconciler.go

```diff
@@ -72,12 +83,48 @@ func (r *TenantInternalReconciler) reconcileTenant() (*porta_client_pkg.Tenant,
 		return nil, err
 	}
 
-	if tenantDef == nil {
+	// if tenant is found in 3scale and deletion timestamp is present, attempt to delete
```

Comment:
I think we could pass this inside a function to ReconcileFinalizer, once it's completed, the finalizer can be removed